pkg/reconciler/knativeserving/common/ha.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	mf "github.com/manifestival/manifestival"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes/scheme"
+	servingv1alpha1 "knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
+)
+
+const (
+	configMapName        = "config-leader-election"
+	enabledComponentsKey = "enabledComponents"
+	componentsValue      = "controller,hpaautoscaler,certcontroller,istiocontroller,nscontroller"
+)
+
+var deploymentNames = sets.NewString(
+	"controller",
+	"autoscaler-hpa",
+	"networking-certmanager",
+	"networking-ns-cert",
+	"networking-istio",
+)
+
+// HighAvailabilityTransform mutates configmaps and replicacounts of certain
+// controllers when HA control plane is specified.
+func HighAvailabilityTransform(instance *servingv1alpha1.KnativeServing, log *zap.SugaredLogger) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if instance.Spec.HighAvailability == nil {
+			return nil
+		}
+
+		// Transform the leader election config.
+		if u.GetKind() == "ConfigMap" && u.GetName() == "config-leader-election" {
+			cm := &v1.ConfigMap{}
+			if err := scheme.Scheme.Convert(u, cm, nil); err != nil {
+				return err
+			}
+
+			if cm.Data == nil {
+				cm.Data = map[string]string{}
+			}
+			cm.Data[enabledComponentsKey] = componentsValue
```

Comment:
You might want to log a warning if this value is also present in the `instance.Config` map... maybe?